azurerm/resource_arm_application_gateway.go

```diff
@@ -146,6 +146,11 @@ func resourceArmApplicationGateway() *schema.Resource {
 							}, true),
 						},
 
+						"host_name": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
```

Comment:
Could we get some validation here? at the very least `validate.NoZeroValues`

Reply:
Since the value is not always set, should I stil add this validation? Which case would this cover?

Reply:
+1 I think since this is Optional NoZeroValues doesn't make sense here